pkg/process/util/ipc.go

```diff
@@ -0,0 +1,49 @@
+package util
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/api/pb"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials"
+)
+
+// GetDDAgentClient creates a pb.AgentClient for IPC with the main agent via gRPC. This call is blocking, so
+// it is up to the caller to supply a context with appropriate timeout/cancel options
+func GetDDAgentClient(ctx context.Context) (pb.AgentClient, error) {
+	// This is needed as the server hangs when using "grpc.WithInsecure()"
+	tlsConf := tls.Config{InsecureSkipVerify: true}
+
+	opts := []grpc.DialOption{
```

Comment:
It would be good to maybe add some debug logging in this function to make it easier to troubleshoot issues creating the gRPC connection